### PR TITLE
Crowleys news 6

### PR DESCRIPTION
**Pull request by WhitworthJoe**

### CODE REVIEW

File: db/app/app-tests/app.test.js [hunk 1]
```diff
@@ -114,3 +114,43 @@ describe('/api/getArticles', () => {
...
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject(
+            {
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              article_id: expect.any(Number),
+            },
```
Comment (2202Hannah) [inline db/app/app-tests/app.test.js:137]: Nice use of jest sorted here! 
Great use of the flexible jest sorter toMatchObject().

**As you are asserting inside a forEach() you should also make sure that the array is not empty.**

File: db/app/app-tests/app.test.js [hunk 2]
```diff
@@ -114,3 +114,43 @@ describe('/api/getArticles', () => {
...
+        expect(body.msg).toEqual("No comments for this article");
+      });
+  });
+  test("400: returns error for invalid article_id", () => {
+    return request(app)
+      .get("/api/articles/hello/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Bad request");
+      });
```
Comment (2202Hannah) [inline db/app/app-tests/app.test.js:165]: Nice work on you error handling tests :)

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hi Joe, great work on this endpoint so far. 

You seem to be missing a 200 test for when an article exists but has no comments. 
I can see that you have considered the functionality for this in your utils.js file already, however this may need tweaking slightly too. If an article exists but has no comments then this would not actually be an error. What would we expect to see returned in the response?

File: db/app/app-tests/app.test.js (hunk 1 above)
Comment (itskatherine) [reply db/app/app-tests/app.test.js:137]: Still need to do this Joe ☝️ 
Can you add an assertion to make sure this array of comments isn't empty

File: db/app/app-tests/app.test.js [hunk 3]
```diff
@@ -114,3 +114,53 @@ describe('/api/getArticles', () => {
...
+  });
+  test('200: should return an empty array when there are no comments', () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const {comments} = body
+        expect(comments).toEqual([])
+      });
```
Comment (itskatherine) [inline db/app/app-tests/app.test.js:149]: Great 👍

File: db/app/controller.js [hunk 4]
```diff
@@ -32,3 +39,18 @@ exports.getArticlesById = (req, res, next) => {
...
+    commentPromises.push(checkExists("articles", "article_id", article_id));
+  }
+  Promise.all(commentPromises)
+    .then((resolvedPromises) => {
+      const comments = resolvedPromises[0];
+      res.status(200).send({ comments });
+    })
+    .catch(next);
+};
+
```
Comment (itskatherine) [inline db/app/controller.js:56]: Great 👍

File: db/app/utils.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const db = require("../../db/connection")
+const format = require("pg-format");
+
+exports.checkExists = (table, column, value) => {
+  const queryString = format(`SELECT * FROM %I WHERE %I = $1;`, table, column);
+  return db.query(queryString, [value]).then(({ rows }) => {
+    if (!rows.length) {
+      return Promise.reject({ status: 404, msg: "No comments for this article" });
+    }
+  });
```
Comment (itskatherine) [inline db/app/utils.js:10]: 👍

**Review by itskatherine — COMMENTED** [review]
Nice work on the 200 test for no comments Joe, still need to check the length on the comments array in your first test but once you've done that you're good to merge 👍

**Pull request merged**